### PR TITLE
fix: only watch lock files in workspace root

### PR DESCRIPTION
**Pull request by ChiefORZ**
Closes https://github.com/biomejs/biome/issues/1601

Screenshot 2024-01-22 at 14 39 54

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -193,8 +193,11 @@ export async function activate(context: ExtensionContext) {
 	await reloadClient();
 
 	// Best way to determine package updates. Will work for npm, yarn, pnpm and bun. (Might work for more files also).
+	// Also ignores .wireit folder, because it's lock file is not relevant and forces a unrelated reload.
 	// It is not possible to listen node_modules, because it is usually gitignored.
-	const watcher = workspace.createFileSystemWatcher("**/*lock*");
+	const watcher = workspace.createFileSystemWatcher(
+		"{**/*lock*,!(**/.wireit/**)}",
```
Comment (ematipico) [inline src/main.ts:199]:
```suggestion
		"{**/*lock*,!**/.wireit/**}",
```

**Comment by nhedger** [comment]
I'm wondering if instead we shouldn't just watch lockfiles in the root of the workspace folder.

**Comment by ChiefORZ** [comment]
@nhedger but then there could be the risk that lockfiles in a multi-root vscode workspace - will not trigger a reload. just a mind of though because i personally work with multi-root workspaces. but im no expert in how vscode works

**Comment by nhedger** [comment]
At this time we do not support multiple workspace folders. We're hoping to add support soon though.

That being said, nothing would prevent us from creating a watcher per workspace folder if we needed to.

The reason I proposed this is that currently the extension will use biome from the dependencies declared at the root of your project anyway, so watching lock files deeper in the hierarchy is basically useless.

**Comment by ChiefORZ** [comment]
@nhedger the PR always fails because in our codebase the "source.organizeImports.biome" does not work like it is described in biome's docs - we need the additional "source.organizeImports" to be enabled too.

```json
{
  "editor.codeActionsOnSave": {
    "source.organizeImports": "explicit",
    "source.organizeImports.biome": "explicit",
  }
}
```

With this codebase there is a race condition, so i have to disable the "source.organizeImports" rule and it works. But then our codebase stops working - do you maybe have any insights in why this is happening?

**Comment by nhedger** [comment]
Gotcha, would you mind filing an issue for this problem?

File: src/main.ts [hunk 2]
```diff
@@ -192,28 +193,32 @@ export async function activate(context: ExtensionContext) {
...
-				await reloadClient();
-				if (client.isRunning()) {
-					await client.restart();
-				} else {
-					await client.start();
+	if (workspace.workspaceFolders?.[0]) {
+		// Best way to determine package updates. Will work for npm, yarn, pnpm and bun. (Might work for more files also).
+		// It is not possible to listen node_modules, because it is usually gitignored.
+		const watcher = workspace.createFileSystemWatcher(
+			new RelativePattern(workspace.workspaceFolders[0], "*lock*"),
```
Comment (nhedger) [inline src/main.ts:200]: We need to use a RelativePattern here, otherwise the watcher won't work as expected, which we want it to watch only the root.

**Comment by nhedger** [comment]
Thanks for you contribution @ChiefORZ!

**Pull request merged**